### PR TITLE
Catroid-752 - Feature flag as condition for RegEx assistant functionalities

### PR DESCRIPTION
**Pull request by UneclStivun**
depends on #3794 
-add feature flag in build.gradle
-add condition for RegEx assistant functionalities via feature flag

https://jira.catrob.at/browse/CATROID-752

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by david-loe — APPROVED** [review]
looks fine to me.

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/dialog/RegularExpressionAssistantDialogTest.java [hunk 1]
```diff
@@ -100,6 +108,34 @@ public void testIsHtmlExtractorInList() {
...
+
+	@Test
+	public void testOpenWikipageOnWikiButtonClick() {
+		Intents.init();
+
+		//setup
+		clickOnAssistantInFunctionList();
+
+		try {
+			//Click Catrobat RegEx Wiki Button
```
Comment (wslany) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/dialog/RegularExpressionAssistantDialogTest.java:128]:
```suggestion
```
Code is clear enough.

File: catroid/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1776,6 +1776,7 @@ needs read and write access to it. You can always change permissions through you
         assistant</string>
     <string name="formula_editor_dialog_regular_expression_assistant_title">Regular
         expression assistant</string>
+    <string name="formula_editor_dialog_wiki_button">Catrobat RegEx Wiki</string>
```
Comment (wslany) [inline catroid/src/main/res/values/strings.xml:1779]:
```suggestion
    <string name="formula_editor_dialog_wiki_button">More information</string>
```

**Review by wslany — CHANGES_REQUESTED** [review]

**Review by M0d0s — APPROVED** [review]

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!

**Pull request merged**